src/api/app/controllers/status_project_controller.rb

```diff
@@ -1,41 +1,68 @@
 class StatusProjectController < ApplicationController
+  before_action :initialize_caches, only: [:show]
+
   def show
     dbproj = Project.get_by_name(params[:project])
     @packages = ProjectStatus::Calculator.new(dbproj).calc_status
-    find_relationships_for_packages(@packages)
+    find_relationships_for_packages
   end
 
   private
 
+  def initialize_caches
+    @package_hash = {}
+  end
+
   def role_from_cache(role_id)
-    @rolecache[role_id] ||= Role.find(role_id).title
+    @rolecache ||= Hash.new do |h, rid|
+      h[rid] = Role.find(rid).title
+    end
+    @rolecache[role_id]
   end
 
   def user_from_cache(user_id)
-    @usercache[user_id] ||= User.find(user_id).login
+    @usercache ||= Hash.new do |h, uid|
+      h[uid] = User.find(uid).login
+    end
+    @usercache[user_id]
   end
 
   def group_from_cache(group_id)
-    @groupcache[group_id] ||= Group.find(group_id).title
+    @groupcache ||= Hash.new do |h, gid|
+      h[gid] = Group.find(gid).title
+    end
+    @groupcache[group_id]
   end
 
-  def find_relationships_for_packages(packages)
-    @usercache = {}
-    @groupcache = {}
-    @rolecache = {}
-    package_hash = {}
-    packages.each_value do |p|
-      package_hash[p.package_id] = p
-      package_hash[p.develpack.package_id] = p.develpack if p.develpack
+  def package_hash(packages)
```

Comment:
We know `find_relationships_for_packages` runs first. We know it will fill `@package_cache`. We know all subsequent calls to `package_hash` will use `@package_cache`. What are we doing here?